src/main/java/com/globalitgeeks/examninja/usermanagement/controller/UserController.java

```diff
@@ -32,7 +32,15 @@ public ResponseEntity<?> register(@RequestBody UserRequest userRequest) {
     }
 
     // Login Endpoint
-
+     @PostMapping("/login")
+    public ResponseEntity<?> login(@RequestBody LoginRequest loginRequest) {
+        try {
+            User user = userService.login(loginRequest.getEmail(), loginRequest.getPassword());
+            return ResponseEntity.ok(user);
+        } catch (Exception e) {
+            return ResponseEntity.status(401).body(e.getMessage());
```

Comment:
Catching generic exceptions (Exception e) is not ideal. It's better to catch specific exceptions related to authentication failures, such as AuthenticationException or custom exceptions like InvalidCredentialsException. This provides more meaningful responses.
If you need more flexibility in error handling, consider using a global exception handler (@ControllerAdvice) to manage exceptions across all controllers.
@PostMapping("/login")
public ResponseEntity<?> login(@RequestBody LoginRequest loginRequest) {
    try {
        User user = userService.login(loginRequest.getEmail(), loginRequest.getPassword());
        return ResponseEntity.ok(user);
    } catch (InvalidCredentialsException e) {
        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("Invalid credentials");
    } catch (Exception e) {
        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("An unexpected error occurred");
    }
}